src/ms/deconvolution.js

```diff
@@ -0,0 +1,49 @@
+import { ngmca } from 'ml-ngmca';
+
+/**
+ * Performing non-negative matrix factorization solving argmin_(A >= 0, S >= 0) 1 / 2 * ||Y - AS||_2^2 + lambda * ||S||_1
+ * @param {Chromatogram} chromatogram - GC/MS chromatogram where make the estimation.
+ * @param {Array<Object>} ranges - List of ranges with from to and optional the sub-matrix with the mass x axis, times, and the number of pure components
+ * @param {Object} options - Options of ngmca factorization method
+ * @param {Number} [options.maximumIteration = 500] - Maximum number of iterations
+ * @param {Number} [options.maximumFBIIteration = 80] - Maximum number of iterations of the Forward-Backward subroutine
+ * @param {Number} [options.FBRelativeDifferenceTolerance = 1e-5] - relative difference tolerance for convergence of the Forward-Backward sub-iterations
+ * @param {Number} [options.phaseRatio = 0.8] - transition between decreasing thresholding phase and refinement phase in pourcent of the iterations
+ * @param {Number} [options.tauMAD = 1] - constant coefficient for the final threshold computation
+ */
+
+export function deconvolution(chromatogram, ranges, options) {
```

Comment:
there seems to be no optimisation made in case of more than one range, so I wouldn't add this in the API. The user can just call the method multiple times if they have multiple ranges

Reply:
yes, I did apply your suggestion